serving/purgatorio/select-and-train-machine-learning-models/tools-and-libraries/tools-and-libraries.md

```diff
@@ -1,5 +1,321 @@
+
+Learning tools and libraries will be a thing most of your life if you want to study or work in the Data Science field. 
+
+**But, as a beginner**:
+
+- is better to start learning very well _some_ tools/libraries, and meanwhile becoming "aware" of the similar ones (and their pro/cons).
+
+- For doing that, it's useful to read the homepage of the official documentation (God bless good documentation!). 
+
+- It's way better to learn **one** (max 2) libraries for each step of the Data Science process, _as a starting point!_
+
+- You'll always be capable of learning more libraries when you reach the capabilities limits of the one that you already know or you have different needs.  
+
+Now you know what is the best approach to learn tools and libraries and frameworks, and how to do it [using wisely your time](https://waitbutwhy.com/2014/05/life-weeks.html)!
+
+The following links expand these points and you should read them if you are a beginner, some answers are illuminating:
+
+- [What are some tips for a beginning ML/data scientist who feels overwhelmed?](https://www.quora.com/What-are-some-tips-for-a-beginning-ML-data-scientist-who-feels-overwhelmed)
+- [SO MUCH to Learn... Overwhelmed. How to Learn so Much?](https://www.reddit.com/r/learnprogramming/comments/dnujl0/so_much_to_learn_overwhelmed_how_to_learn_so_much/)
+- [Beginner to "data science" but unsure of what I should be doing](https://www.reddit.com/r/datascience/comments/7yg61i/beginner_to_data_science_but_unsure_of_what_i/)
+
+Having that said, let's jump in!
+
+# The Data Science Process Toolkit
+## Data Collection
+
+Data collection is one of the most crucial parts of a Data Science project (you don't say?), and often a tedious one if you have to label your data on your own!
+
+Luckily, good researchers and institutions often release their datasets, so you don't have to collect the data (or label it), just download!
+
+But if you can't find the dataset that you need? 
+
+Let's see at some tools that can help you in this task:
+
+- [Digitalmethods Tools](https://wiki.digitalmethods.net/Dmi/ToolDatabase) is a set of free web scraping tools developed by Europe's leading Internet Studies groups
+
+- [BeautifulSoup](https://www.dataquest.io/blog/web-scraping-tutorial-python/) is a Python library to scrape webpages
+
+- [Selenium](https://www.pluralsight.com/guides/web-scraping-with-selenium) is a Python library meant to automate browser testing, but it can be perfectly used in combination with BeautifulSoup to set up a powerful web scraper
+
+See also:
+- [10 Best Web Scraping Tools to Extract Online Data](https://www.hongkiat.com/blog/web-scraping-tools/)
+
+Remember that a lot of free access datasets are available for free, for example in the [UCI Dataset](https://archive.ics.uci.edu/ml/datasets.php) website or at the Data science Challenge platform [Kaggle](https://www.kaggle.com/).
```

Comment:
I can suggest this link as well, for [Data Collections](https://github.com/neomatrix369/awesome-ai-ml-dl/blob/master/data/datasets.md)

Reply:
uh very nice I add this